src/helpers/file_helpers.h

```diff
@@ -3,146 +3,55 @@
-   * expected substring is 6 consecutive "X" characters (i.e. XXXXXX). Users may specify their own
-   * replacement string. Note that only the _first_ matching substring is replaced. If the
-   * replaceToken is not found, then the templateString is returned, unmodified.
-   * Each replacement letter may either be upper or lower case.
-   * Similar to what QTemporaryFile does.
-   * @param templateStr The model string, with
-   * @param replaceToken The template string to find, and then replace, with random characters
-   * @return The resulting filename
+   * @brief randomString Generates a random String of N chars
+   * Each character can be a-z either be upper or lower case.
+   * @param numberOfChars Length of the string to return default is 6
+   * @return The resulting randomString
    */
-  static QString randomFilename(QString templateStr, QString replaceToken="XXXXXX") {
-    int templateIndex = templateStr.indexOf(replaceToken);
-
-    if (templateIndex == -1) {
-      return templateStr;
-    }
-
-    QString replacement = randomText(replaceToken.length());
-    return templateStr.replace(templateIndex, replaceToken.length(), replacement);
-  }
-
-  /// writeFile write the provided content to the provided path.
-  /// @throws a FileError if there are issues opening or writing to the file.
-  static void writeFile(QString path, QString content) {
-    writeFile(path, content.toUtf8());
+  static QString randomString(int numberOfChars = 6) {
+    QString rString;
+    for(int i = 0; i < numberOfChars; i++)
+        rString.append(_chars.at(QRandomGenerator::global()->bounded(_chars.length())));
+    return rString;
   }
 
   /// writeFile write the provided content to the provided path.
-  /// @throws a FileError if there are issues opening or writing to the file.
-  static void writeFile(QString path, QByteArray content) {
-    QFile file(path);
-    bool opened = file.open(QIODevice::WriteOnly);
-    if (opened) {
-      int bytesWritten = file.write(content);
-      if (bytesWritten == -1) {
-        throw FileError::mkError("Error writing file", path.toStdString(), file.error());
-      }
-      file.close();
-    }
-    if (file.error() != QFile::NoError) {
-      throw FileError::mkError("Unable to write to file", path.toStdString(), file.error());
-    }
+  /// returns false if failed.
+  static bool writeFile(QString fileName, QByteArray content) {
+    auto filePath = getDirname(fileName);
+    if(!QDir().exists(filePath))
+        QDir().mkpath(filePath);
+    QFile file(fileName);
+    if (!file.open(QIODevice::WriteOnly))
+        return false;
+    if (file.write(content) == -1)
+        return false;
+    return true;
   }
 
   /// readFile reads all of the data from the given path.
-  /// @throws a FileError if any issues occur while writing the filethere are issues opening or
+  /// returns empty QByteArray if unable to read.
```

Comment:
We should revise this comment:

```suggestion
  /// returns empty QByteArray if unable to read or if there is no data in the file. This is a limitation of QFile's readAll method
```